lib/ruby_smb/dcerpc/winreg/query_value_request.rb

```diff
@@ -24,10 +25,10 @@ def initialize_instance
           @opnum = REG_QUERY_VALUE
         end
 
-        # Determines the correct length for the padding in front of
-        # #lp_type. It should always force a 4-byte alignment.
-        def pad_length
-          offset = (lp_value_name.abs_offset + lp_value_name.to_binary_s.length) % 4
+        # Determines the correct length for the padding, so that the next
+        # field is 4-byte aligned.
+        def pad_length(prev_element)
+          offset = (prev_element.abs_offset + prev_element.to_binary_s.length) % 4
           (4 - offset) % 4
```

Comment:
The modulo 4 here isn't necessary because `offset` is already bound to 0-3 due to the modulo 4 operator in it's assignment.

Reply:
Same here, if `prev_element.abs_offset + prev_element.to_binary_s.length` is already 4-byte aligned, `offset` value will be 0, which will result in an unnecessary 4-byte pad. Adding a second modulo operation will ensure the pad will be 0-length in this scenario.